### PR TITLE
DonoG-123

### PR DESCRIPTION
**Pull request by deva9**
JIRA tickets: 123, 124
Integrated payment portal.
Saves user interests to database.
Minor UI and bug fixes

### CODE REVIEW

File: DonoGear/app/src/main/java/com/example/donogear/actionpages/PaymentActivity.java [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+        initializeLayout();
+        purchase.setOnClickListener(v -> buy());
+        shipping.setOnClickListener(view -> {
+            card = cardMultilineWidget.getCard();
+            startActivity(new Intent(getApplicationContext(), ShippingDetailsActivity.class));
+        });
+        reset.setOnClickListener(view -> cardMultilineWidget.clear());
+    }
+
+    private void initializeLayout() {
```
Comment (elegier) [inline DonoGear/app/src/main/java/com/example/donogear/actionpages/PaymentActivity.java:78]: missing javadoc

File: DonoGear/app/src/main/java/com/example/donogear/actionpages/PaymentActivity.java [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+            purchase.setAlpha(0.25f);
+        } else {
+            purchase.setAlpha(1f);
+        }
+        super.onResume();
+    }
+
+    /**
+     * Validate if user has entered all fields
+     * @return
```
Comment (elegier) [inline DonoGear/app/src/main/java/com/example/donogear/actionpages/PaymentActivity.java]: missing return comment

File: DonoGear/app/src/main/java/com/example/donogear/actionpages/PaymentActivity.java [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+     * Progress dialog for UI
+     * @param title - title for dialog
+     */
+    private void startProgress(String title){
+        progress.setTitle(title);
+        progress.setMessage("Please Wait");
+        progress.show();
+    }
+
+    private void finishProgress(){
```
Comment (elegier) [inline DonoGear/app/src/main/java/com/example/donogear/actionpages/PaymentActivity.java:230]: missing javadoc

File: DonoGear/app/src/main/java/com/example/donogear/utils/Constants.java [hunk 4]
```diff
@@ -42,12 +44,30 @@
 
     public static final String ERROR_BID_TITLE = "Error placing your bid";
     public static final String ERROR_BID_MESSAGE =
-            "The most recent bid amount was changed when you were trying to bid. Do you want to " +
+            "The most recent bid amount was changed while you were trying to bid. Do you want to " +
```
Comment (elegier) [inline DonoGear/app/src/main/java/com/example/donogear/utils/Constants.java]: I think the previous message was better here

**Review by elegier — CHANGES_REQUESTED** [review]
address comments and then merge

**Comment by MeetChheda** [comment]
I have worked upon all the changes

**Pull request merged**